src/rollup-preset/index.ts

```diff
@@ -0,0 +1,248 @@
+import {readFileSync} from 'node:fs';
+import {dirname, resolve} from 'node:path';
+import {cwd} from 'node:process';
+
+import {
+  loadPartialConfig,
+  PluginItem,
+  PluginOptions,
+  TransformOptions,
+} from '@babel/core';
+import {getBabelOutputPlugin} from '@rollup/plugin-babel';
+import image from '@rollup/plugin-image';
+import typescript from '@rollup/plugin-typescript';
+import {RollupOptions} from 'rollup';
+
+/**
+ * Modifications to be applied to the Babel configuration.
+ *
+ * This object defines modifications to Babel plugins and presets.
+ * - Plugins can be removed by setting their corresponding value to `null`.
+ * - Presets can be modified by providing a function that updates their options.
+ */
+const BABEL_MODIFICATIONS = {
+  plugins: {
+    /**
+     * Removes the `module-resolver` plugin from the Babel configuration.
+     *
+     * Setting the value to `null` indicates that the plugin should be removed.
+     */
+    'module-resolver': null,
+  },
+  presets: {
+    /**
+     * Updates the options for the `@react-native/babel-preset` preset.
+     *
+     * This modification function disables the `enableBabelRuntime` option in the preset.
+     *
+     * @param options - The existing options for the preset.
+     * @returns The modified options with `enableBabelRuntime` set to `false`.
+     */
+    '@react-native/babel-preset': (options: any) => ({
+      ...options,
+      enableBabelRuntime: false,
+    }),
+  },
+};
+
+/**
+ * Updates or removes plugins/presets from the Babel configuration.
+ *
+ * @param babelConfig - The loaded Babel configuration.
+ * @param modifications - An object defining plugins/presets to remove or update.
+ * @returns The modified Babel configuration.
+ */
+function modifyBabelConfig(
+  babelConfig: TransformOptions | null,
+  modifications: {
+    plugins?: {
+      [key: string]: null | ((options: PluginOptions) => PluginOptions);
```

Comment:
🤔 maybe move it in a separate type? occurs several times in this file

Reply:
yeah good point

Reply:
updated